interfaces/mount/backend.go

```diff
@@ -65,6 +65,7 @@ func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementO
 	}
 	spec.(*Specification).AddOvername(snapInfo)
 	spec.(*Specification).AddLayout(snapInfo)
+	spec.(*Specification).AddExtraLayouts(confinement.ExtraLayouts)
```

Comment:
layout need also extra apparmor rules to set up things for snap-update-ns to be able to perform them. So you need similar changes also to interfaces/appamor,  you will see it has as a AddLayout too.

Reply:
Thanks! I'll look into this

Reply:
I have added support in apparmor for adding the additional layouts